exercises/bowling/example.go

```diff
@@ -0,0 +1,126 @@
+// Package bowling implements scoring for the game of bowling.
+package bowling
+
+import "errors"
+
+const testVersion = 1
+
+var (
+	ErrNegativeRollIsInvalid        = errors.New("Negative roll is invalid")
+	ErrPinCountExceedsPinsOnTheLane = errors.New("Pin count exceeds pins on the lane")
+	ErrPrematureScore               = errors.New("Score cannot be taken until the end of the game")
+	ErrCannotRollAfterGameOver      = errors.New("Cannot roll after game is over")
+)
+
+// Game records the data to track a game's progress.
+type Game struct {
+	rolls       [21]int // storage for the rolls (10th frame gets up to 3)
+	nRolls      int     // counts the rolls accumulated.
+	nFrames     int     // counts completed frames, up to 10.
+	rFrameStart int     // tracks the starting roll of each frame.
+}
+
+// NewGame returns a fresh zero-valued game struct.
+func NewGame() *Game {
+	return &Game{}
+}
+
+// Roll records one roll for a bowling frame with 'pins' knocked down.
+// An error is possible depending on pin value and previous rolls.
+func (g *Game) Roll(pins int) error {
+	// Validate pin count on roll.
+	if pins > 10 {
```

Comment:
a lot of magic numbers in here, could they become consts such as `pinsPerFrame`? I know they'll never change so it we don't get the advantage of "easily be able to change the constant if we need to", but we do get the advantage of "what exactly does this number mean and why is it 10?" - and this is doubly important because there are multiple 10s in this code and some of them "number of frames in a game" and some of them are "number of pins in a frame"